xapian-core/tests/harness/testsuite.cc

```diff
@@ -881,7 +881,12 @@ test_driver::parse_command_line(int argc, char **argv)
     if (verbose == 0) {
 	const char *p = getenv("VERBOSE");
 	if (p != NULL) {
-	    verbose = atoi(p);
+	    if (!parse_signed(p, verbose) || (verbose != 1 && verbose != 2)) {
+		throw "Verbose must be 1 or 2"
+		      "a value of 1 will display detailed information about"
+		      "failures, and a value of 2 will display further"
+		      "information about the progress of tests.";
+	    }
```

Comment:
`VERBOSE=0` currently has the same effect as an empty value or not setting `VERBOSE`.

I tend to think it's better allow any non-negative integer value here - at some point increasing the value won't actually make things more verbose but it's handy to be able to type `VERBOSE=999` to get maximum verbosity rather than have to remember (or look up) what the current maximum value actually is.  And the Python bindings testsuite is more verbose for `VERBOSE=3` than `VERBOSE=2` and so running `VERBOSE=3 make check` from the top-level really ought to work.

Also you're missing a space at each break point so the joined string will say "Verbose must be 1 or 2a value", etc.

Reply:
fixed in a fixup commit.